Refit/RequestBuilderImplementation.cs

```diff
@@ -32,23 +33,80 @@ public RequestBuilderImplementation(Type targetInterface, RefitSettings refitSet
             }
 
             targetType = targetInterface;
-            interfaceHttpMethods = targetInterface.GetMethods()
-                                                  .Where(x => x.GetCustomAttributes(true)
-                                                               .OfType<HttpMethodAttribute>().Any())
-                                                  .Select(x => new RestMethodInfo(targetInterface, x, settings))
-                                                  .ToDictionary(k => k.Name, v => v);
+
+            foreach (var methodInfo in targetInterface.GetMethods()) {
+                var attrs = methodInfo.GetCustomAttributes(true);
+                var hasHttpMethod = attrs.OfType<HttpMethodAttribute>().Any();
+                if (hasHttpMethod) {
+                    var restinfo = new RestMethodInfo(targetInterface, methodInfo, settings);
+                    interfaceHttpMethods.AddOrUpdate(methodInfo.Name, s => new[] { restinfo }.ToList(), (s, list) => {
```

Comment:
I'm not sure what the concurrency guarantees are trying to be here. On one side there's a concurrent dictionary but then inside there's a regular list. If the concern is that the generator method may be called on different threads at the same time, this could lead to duplicate items in the list even if the keys are the same.

Perhaps it would be easier to use a local Dictionary instance inside of the method to populate, then set the class member to the instance after it's populated? Worst case is that the two different dictionaries are generated but they're functionally the same and have the same content. Last one overwrites the other, but the overwrite of the field reference to the dictionary is atomic.